rust/lancedb/src/catalog/listing.rs

```diff
@@ -0,0 +1,560 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The LanceDB Authors
+
+//! Catalog implementation based on a local file system.
+
+use std::collections::HashMap;
+use std::fs::create_dir_all;
+use std::path::Path;
+use std::sync::Arc;
+
+use super::{
+    Catalog, CreateDatabaseMode, CreateDatabaseRequest, DatabaseNamesRequest, OpenDatabaseRequest,
+};
+use crate::connection::ConnectRequest;
+use crate::database::listing::ListingDatabase;
+use crate::database::Database;
+use crate::error::{CreateDirSnafu, Error, Result};
+use async_trait::async_trait;
+use lance::io::{ObjectStore, ObjectStoreParams, ObjectStoreRegistry};
+use lance_io::local::to_local_path;
+use lance_io::object_store::WrappingObjectStore;
+use object_store::path::Path as ObjectStorePath;
+use snafu::ResultExt;
+
+#[derive(Debug)]
+pub struct ListingCatalog {
+    object_store: ObjectStore,
+
+    query_string: Option<String>,
+
+    pub(crate) uri: String,
+
+    base_path: ObjectStorePath,
+
+    pub(crate) store_wrapper: Option<Arc<dyn WrappingObjectStore>>,
+
+    storage_options: HashMap<String, String>,
```

Comment:
The `storage_options` that get passed in through `connect` are already used to create `object_store` but not used elsewhere.  Do we want to also pass these `storage_options` in to the database when we open / create a database?

Same question for `store_wrapper`.

Reply:
Actually, I am not sure if we get an instance of `ObjectStore`, do we still need `storage_options` and `store_wrapper` in known scenario?